keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1237 @@
+      to allow `defaultRequest` and `default` in Pod LimitRange object.
+
+      * Conditional logic
+        
+         * Pod level resource spec is set and `Pod` LimitRanger is applied.
+           If pod level resources are specified, use these values directly against the Pod type LimitRanger. It doesn't matter if container level
+           resource spec is set or not.
+           
+         * Pod level resource spec is set and `Pod` LimitRanger is applied.
+           If pod level resources are not specified, there could be cases:
+             1. Existing workloads that have `Pod` LimitRanger set to validate the constraints at container level aggregates (and not at pod level).
+             2. New workloads that want to use Pod Level resource limits feature and have set `Pod` LimitRanger to validate constraints at Pod Level.
+           
+           If the behavior of `Pod` LimitRanger is modified, it could break existing workloads and also remove the ability for usecase where users 
+           want to have limit range constraints for container level aggregates.         
+
+  * **Option 2: Add a new LimitRanger type PodLevelResources**
+    * Introduce a new type `PodLevelResources` in the LimitRange specifically for defining pod-level resource constraints (min, max, defaultRequest, default, maxLimitRequestRatio).
+  
+    This new type would be independent of the existing Pod and Container LimitRange types with clearer semantics. It maintains backward compatibility as it doesn't modify
+    existing Pod type. User education will be needed to clarify the difference between Pod and PodLevelResources types. Plus this is confusing with Pod type LimitRange.
+
+  * **[Preferred] Option 3: Add a new optional field `Scope` in LimitRangeSpec**
+    * Introduce a new optional field in LimitRangeSpec with values: Pod or Container(default).
+
+      ```
+        // LimitRangeSpec defines a min/max usage limit for resources that match on kind
+        type LimitRangeSpec struct {
+	      
+          // Limits is the list of LimitRangeItem objects that are enforced
+	        Limits []LimitRangeItem
+          // Allowed values: "Container" (default) or "Pod"
+          Scope LimitRangeScope `json:"scope,omitempty"
+        }
+      ```
+      
+      The validation logic will use Pod level resource spec directly if scope is set to Pod, otherwise it will apply existing logic for aggregated container resources.
+
+
+    * Support `default` and `defaultRequest` constraints if scope is "Pod".  
+
+    This option maintains backward compatibility and offers flexibility, allowing both pod-level and container-level constraints in the same LimitRange object. The implementation
+    is much simpler than Option 2. Although some user education through docs is still required with this option.
+  
+
+* ResourceQuota
```

Comment:
Similar to above, this whole section feels like it could be reduced to something like "Quota will consider the effective resource requests and limits for the pod, including the new pod-level resources." ?